docs/source/gui.rst

```diff
@@ -228,15 +228,98 @@ of the configuration. Information about the avilable properties can be found in
 Plasma Shape Editor
 -------------------
 
-TODO: The plasma shape editor is currently still under development. The
-documentation will be updated once it is in a more finished state.
+The Plasma Shape Editor provides an interactive interface for running the 
+`NICE <https://gitlab.inria.fr/blfauger/nice>`__ (Newton direct and Inverse Computation 
+for Equilibrium) solver. This allows you to design and analyze plasma equilibria by 
+either specifying a desired shape and finding the necessary coil currents (Inverse Mode) 
+or by providing coil currents and calculating the resulting plasma shape (Direct Mode).
 
-p' and ff' Parameterizations
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The interface is primarily divided into two sections: a large plotting area on the left 
+that displays the machine geometry and equilibrium results, and a control panel on the 
+right with various configuration options.
+
+.. figure:: images/gui/plasma_shape_editor.png
+
+   Overview of the Plasma Shape Editor
+
+Operating Modes
+^^^^^^^^^^^^^^^
+
+NICE can be run in two modes, which can be selected using the radio buttons at the top 
+of the control panel:
+
+* **Inverse Mode**: You define a target plasma shape and properties, and NICE calculates 
+  the required coil currents to achieve this equilibrium. You can also choose to 
+  fix the current in a subset of coils.
+* **Direct Mode**: You specify the currents for all coils, and NICE 
+  calculates the resulting plasma equilibrium.
+
+At the top of the control panel, you'll find **Run** and **Stop** buttons to start and 
+terminate the NICE simulation. A terminal window below these buttons displays live 
+output from NICE.
+
+Configuration Options
+^^^^^^^^^^^^^^^^^^^^^
+
+The control panel on the right contains several collapsible option menus, each dedicated 
+to a specific part of the configuration. A warning icon (⚠️) next to a option title 
+indicates that required information for this section is missing, and needs to be updated 
+for NICE to run.
+
+The following option menus are available, these will be explained in detail in the 
+sections below:
+
+* **NICE Configuration**: Required NICE configuration parameters.
+* **Plotting Parameters**: Configuration for the plot.
+* **Plasma Shape**: Defining a target plasma shape when running NICE in **Inverse Mode**.
+* **Plasma Properties**: Plasma's physical characteristics (e.g. Ip, p' / ff' profiles).
+* **Coil Currents**: Setting the values of the coil currents.
 
-NICE can use a simplified parameterization for the p' and ff' profiles. These can be 
-used by selecting the `Manual` mode in the `Plasma Properties` options menu of the Plasma Shape Editor.
-The following parameterizations are used:
+NICE Configuration
+""""""""""""""""""
+This is where you set up the core parameters for NICE and provide the necessary machine description files.
+
+This section includes settings for:
+
+* The file paths to the NICE executables for both *inverse* and *direct* modes.
```

Comment:
Can you give some pointers for users to determine what they should put here? Would be good to cross-ref with the installation guide for nice